src/Alpacc/RegularExpression.hs

```diff
@@ -0,0 +1,68 @@
+module Alpacc.RegularExpression (regExFromText, pRegEx) where
+
+import Data.Composition
+import Data.Text (Text)
+import Data.Void
+import Text.Megaparsec
+import Text.Megaparsec.Char (char, space1, string)
+import Text.Megaparsec.Char.Lexer qualified as Lexer
+
+type Parser = Parsec Void Text
+
+data RegEx
+  = Epsilon
+  | Literal Char
+  | Star RegEx
+  | Alter RegEx RegEx
+  | Concat RegEx RegEx
+  deriving (Eq, Show)
+
+space :: Parser ()
+space = Lexer.space space1 (Lexer.skipLineComment "#") empty
```

Comment:
Do you really want line comments in your regular expressions?

Reply:
Tbh I did not really think about it, I think comments in RegEx might be nice since then you can explain complex RegEx?

I have seen it often done in python.
https://stackoverflow.com/questions/20668996/commenting-regular-expressions-in-python

Reply:
That's not relevant here.  These regular expressions will ultimately be embedded in a CFG.  They will not be parsed on their own.

Reply:
True this is very silly, I have removed them now by using `empty` I think.